rtt_ros2_params/src/rtt_ros2_params_service.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Intermodalics BVBA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <utility>
+
+#include "rtt/deployment/ComponentLoader.hpp"
+#include "rtt/internal/GlobalService.hpp"
+#include "rtt/plugin/ServicePlugin.hpp"
+
+#include "boost/make_shared.hpp"
+
+#include "rtt_ros2_params/rtt_ros2_params.hpp"
+
+namespace rtt_ros2_params
+{
+
+static bool loadGlobalROSService()
+{
+  // Check and load the ROS 2 service
+  if (!RTT::internal::GlobalService::Instance()->hasService("ros")) {
+    RTT::ComponentLoader::Instance()->import("rtt_ros2", "");
+    RTT::internal::GlobalService::Instance()->provides("ros");
+    if (!RTT::internal::GlobalService::Instance()->hasService("ros")) {
+      RTT::log(RTT::Error) <<
+        "ros service (provided in rtt_ros2) is missing and it is needed "
+        "by rosparam" << RTT::endlog();
+      return false;
+    }
+  }
+
+  auto ros = RTT::internal::GlobalService::Instance()->provides("ros");
```

Comment:
Just copy the code from `rtt_ros2_node`, please. `rtt_ros2` is not needed. Just any service that is loaded first implicitly creates the `"ros"` service by calling `RTT::internal::GlobalService::Instance()->provides("ros")`:

```suggestion
  RTT::Service::shared_ptr ros =
    RTT::internal::GlobalService::Instance()->provides("ros");
  ros->doc("ROS operations and services");
```

Reply:
No service plugin should import another package! I am quite sure that this would cause dead-locks or has other nasty side effects if `rtt_ros2` would not have been imported before (`RTT::ComponentLoader` and `RTT::plugin::PluginLoader` member functions would need to be reentrant).

Normally the `ros.import()` operation provided by `rtt_ros2` should be used to import ROS packages and all its RTT plugin dependencies in the correct topological order. But under no circumstances should a plugin refuse to load or imply a certain order in which plugins are loaded or packages are imported.